enrollment-server/src/main/java/com/wultra/app/enrollmentserver/impl/service/PowerAuthActivationCodeHandler.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * PowerAuth Enrollment Server
+ * Copyright (C) 2025 Wultra s.r.o.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.wultra.app.enrollmentserver.impl.service;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.wultra.app.enrollmentserver.errorhandling.ActivationCodeException;
+import com.wultra.security.powerauth.client.model.entity.ApplicationConfigurationItem;
+import com.wultra.security.powerauth.client.model.error.PowerAuthClientException;
+import com.wultra.security.powerauth.client.model.response.GetApplicationConfigResponse;
+import com.wultra.security.powerauth.client.v4.PowerAuthClient;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Specialization of {@link DelegatingActivationCodeHandler} using application configuration from PowerAuth Server.
+ *
+ * @author Lubos Racansky, lubos.racansky@wultra.com
+ */
+@AllArgsConstructor
+@Slf4j
+class PowerAuthActivationCodeHandler implements DelegatingActivationCodeHandler {
+
+    private static final String ACTIVATION_CODE = "activation_code";
```

Comment:
I find this confusing, because this terminology conflicts with activation using activation code.

Reply:
True, renamed.